core/src/main/java/org/elasticsearch/index/query/InnerHitBuilder.java

```diff
@@ -155,8 +161,8 @@ public InnerHitBuilder(StreamInput in) throws IOException {
         explain = in.readBoolean();
         version = in.readBoolean();
         trackScores = in.readBoolean();
-        fieldNames = (List<String>) in.readGenericValue();
-        fieldDataFields = (List<String>) in.readGenericValue();
+        storedFieldNames = (List<String>) in.readGenericValue();
```

Comment:
We have `readStringArray` in the reader so maybe we should do `new ArrayList<>(in.readStringArray());`. Now is the time to break wire compatibility if we ever do.